### ISSUE TITLE
Provide a way to derive(Bake) on generic types without adding T: Bake bounds

### ISSUE THREAD

**Issue by Manishearth** [unicode-org/icu4x#3624]
A bunch of types (ZeroVec, VarZeroVec, etc) have `impl<T> Bake for Foo<T>` with no additional bounds, as they bake to a byte stream.

It should be possible to opt out of such bounds in derives containing such types, like `CodePointTrie`

**Comment by robertbastian** [comment]
I added the bounds on ZV on purpose, as I'd like to improve the format in the future (the `zerovec!` macro is a step in this direction).

In general it should be possible to opt out though, for example `PhantomData<T>` can be baked without `T: Bake`.

**Comment by sffc** [comment]
Discuss with:

- @Manishearth 
- @sffc 
- @robertbastian

**Comment by Manishearth** [comment]
Yeah, I agree with @robertbastian . In that case we probably need to just add the feature for phantomdata and such.

(unsure if we actually do need to discuss any more)
